### ISSUE TITLE
docs: Release Process

### ISSUE THREAD

**Issue by azu** [textlint/textlint#439]
From #438

We want to create a document for publishing modules.
It is not systematization yet, but want to do systematization.

Currently, I(@azu)'ve released new version at  manually.

## Task

Big picture.

- Bump version
  - Current: `lerna publish` does it
- Update dependencies's version
  - Current: `lerna publish` does it
- Update CHANGELOG
  - Current: `lerna publish --conventional-commits` does it
- Update Release Notes
  - Manually

## Current Workflow

Summarize current workflow

- `yarn run publish`
- Collect CHANGELOG.md (I've used scripts)
- Update GitHub Releases notes

## Current Issue

Current workflow has a risk.

- `yarn check` break current workflow #386 
- `lerna publish--conventional-commits` depended on commit message
  - It is usefule, But It may cause false-positive
  - For example, bump unnessary major update
- Collect CHANGELOG.md is not automated 
- GitHub Releases notes is not useful https://github.com/textlint/textlint/issues/434#issuecomment-354545260
- No dry-run

I prefer safe than fast in this project releasing.

----

TODO

- [x] `release` scripts #492 
- [x] Try to use `release` script
- [ ] Add `git push --tags` at last
- [ ] Add `.github/release.md` for PR template
- [ ] Convert existing release notes to blog article 
  - How to do prettier it?

**Comment by azu** [comment]
We already have a blog.
I will try to think new release process.

**Comment by azu** [comment]
It is difficult to customize lerna workflow.
https://github.com/lerna/lerna/issues/961

**Comment by azu** [comment]
I will add `release.md` as a PR template

[Multiple issue and pull request templates](https://github.com/blog/2495-multiple-issue-and-pull-request-templates "Multiple issue and pull request templates")

**Comment by azu** [comment]
I've implemented initial release scripts #492

**Comment by azu** [comment]
I've tried current release script in #493 
It work.

**Comment by azu** [comment]
📝 https://github.com/GoogleChrome/lighthouse/blob/master/docs/releasing.md

**Comment by azu** [comment]
Notes: I've tried similar process in other project.
It is based on textlint release process.
I'll create to common module to release monorepo in near future.

**Comment by azu** [comment]
https://github.com/azu/monorepo-utils
I've create helper scripts.

**Comment by azu** [comment]
```
yarn run versionUp
git tag --points-at HEAD | xargs -I{} monorepo-utils-collect-changelog {}
git push
# merge
git push --follow-tags

```

**Comment by azu** [comment]
Workflow:

```
# checkout 
git checkout -b {release-name}
# push current branch 
git push origin HEAD -u
# bump vesrion and changelog
yarn run versionup
# git push
git push
# Review it
# After review
yarn run publish
```

After published, merge the review pull request.

**Comment by azu** [comment]
I'll try to add it to CONTRIBUTING.md

**Comment by azu** [comment]
https://github.com/textlint/textlint/blob/master/docs/CONTRIBUTING.md#release-flow
